.github/workflows/test-generation.yml

```diff
@@ -0,0 +1,66 @@
+name: Test PDF & EPUB Generation
+
+on:
+  push:
+    branches: ["main"]
+
+# permissions:
+#   contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        volume: [1, 2, 3]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          if [ -f Scripts/requirements.txt ]; then pip install -r Scripts/requirements.txt; fi
```

Comment:
This has been moved out of the Scripts folder, now it's just `requirements.txt` (because now that it's being used for for the formatter, which didn't really feel like it fit in the Scripts folder)